### PR TITLE
Send button is active when message has spaces, but does not send it

### PR DESCRIPTION
**Pull request by mollfpr**
### Details

1. Add `this.updateComment(this.comment);` to `componentDidMount` so the `comment` state will be check and update the `isCommentEmpty` state, if empty the button will be disable after the page load
2. `trim()` everytime `updateComment()` called and removes whitespace from both sides strings
3. `this.textInput.selectionStart === 0` only to show trigger box edit if cursor position at start position

### Fixed Issues

$ https://github.com/Expensify/App/issues/7965

### Tests

1. Open a chat
2. Verify that the button state is not active if messages on input text is empty
3. Try to add space or newline and verify the button state is not active
4. [Web/Desktop] Try to press the key arrow up and verify the edit box is triggered
5. [Web/Desktop] Try to add spaces or newline, press key arrow up and verify if cursor position at start and arrow up key pressed once more will trigger edit box

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I verified the PR has a small number of commits behind `main`
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I clearly indicated the environment tests should be run in (Staging vs Production)
- [x] I wrote testing steps that cover success & fail scenarios (if applicable)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests & veryfy they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors related to changes in this PR
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I added comments when the code was not self explanatory
    - [x] I put all copy / text shown in the product in all `src/languages/*` files (if applicable)
    - [x] I followed proper naming convention for platform-specific files (if applicable)
    - [x] I followed style guidelines (in [`Styling.md`](https://github.com/Expensify/App/blob/main/STYLING.md)) for all style edits I made
    - [x] I followed the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs))
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I corroborated the UI performance was not affected (the performance is the same than `main` branch)
- [x] If I created a new component I verified that a similar component doesn't exist in the codebase

#### PR Reviewer Checklist
- [ ] I verified the PR has a small number of commits behind `main`
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the testing environment is mentioned in the test steps
- [x] I verified testing steps cover success & fail scenarios (if applicable)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors related to changes in this PR
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified comments were added when the code was not self explanatory
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files (if applicable)
    - [x] I verified proper naming convention for platform-specific files was followed (if applicable)
    - [ ] I verified [style guidelines](https://github.com/Expensify/App/blob/main/STYLING.md) were followed
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components are not impacted by changes in this PR (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified the UI performance was not affected (the performance is the same than `main` branch)
- [x] If a new component is created I verified that a similar component doesn't exist in the codebase

### QA Steps

1. Open a chat
2. Verify that the button state is not active if messages on input text is empty
3. Try to add space or newline and verify the button state is not active
4. [Web/Desktop] Try to press the key arrow up and verify the edit box is triggered
5. [Web/Desktop] Try to add spaces or newline, press key arrow up and verify if cursor position at start and arrow up key pressed once more will trigger edit box

- [x] Verify that no errors appear in the JS console

#### Web

https://user-images.githubusercontent.com/25520267/157205074-ba718970-ff9a-4474-bf45-39f671ed7c04.mov

#### Mobile Web

https://user-images.githubusercontent.com/25520267/157204436-82b97d72-55e7-4601-b00e-b9329434503a.mov

#### Desktop

https://user-images.githubusercontent.com/25520267/157204798-fb03a9cb-d54a-4b0a-b9b9-d7dbd96a1d23.mov

https://user-images.githubusercontent.com/25520267/157204870-8be00df5-ae8d-47d4-9ada-cf2640b92a78.mov

#### iOS

https://user-images.githubusercontent.com/25520267/157204629-0b1b8338-aed5-49d6-ae5d-d642680d23ab.mov

#### Android

https://user-images.githubusercontent.com/25520267/157204517-ab13839f-4c64-43f7-a9da-807b40e52c0d.mov

### CODE REVIEW

**Comment by mollfpr** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by mollfpr** [comment]
recheck

File: src/pages/home/report/ReportActionCompose.js [hunk 1]
```diff
@@ -149,6 +149,8 @@ class ReportActionCompose extends React.Component {
 
             this.focus(false);
         });
+
```
Comment (rushatgabhane) [inline src/pages/home/report/ReportActionCompose.js:152]: @mollfpr remove extra line breaker :)

File: src/pages/home/report/ReportActionCompose.js [hunk 2]
```diff
@@ -346,7 +348,7 @@ class ReportActionCompose extends React.Component {
         }
 
         // Trigger the edit box for last sent message if ArrowUp is pressed
```
Comment (rushatgabhane) [inline src/pages/home/report/ReportActionCompose.js:350]:
```suggestion
        // Trigger the edit box for last sent message if ArrowUp is pressed and the comment is empty
```

**Comment by NikkiWines** [comment]
@mollfpr since this PR is 49 commits behind Expensify:main, please merge main into your branch as well.

**Comment by mollfpr** [comment]
@NikkiWines @rushatgabhane Done!

**Comment by rushatgabhane** [comment]
@mollfpr Thanks for making the changes. I'd suggest you to use more meaningful commit messages for future commits.

eg: instead of "resolve request changes", you could have used "remove extra line and update comment"

**Comment by mollfpr** [comment]
@NikkiWines @rushatgabhane  I will create new PR to give cleaner commit and sign key to my commit. Thanks!

**Pull request closed without merging**